tests/test_auth.py

```diff
@@ -0,0 +1,196 @@
+from unittest import mock
+
+import jwt
+import jwt.utils
+import pytest
+
+from rain_api_core.auth import JwtManager, UserProfile
+
+MODULE = 'rain_api_core.auth'
+
+
+@pytest.fixture
+def jwt_manager(jwt_priv_key, jwt_pub_key):
+    return JwtManager(
+        algorithm='RS256',
+        public_key=jwt_pub_key,
+        private_key=jwt_priv_key,
+        cookie_name='auth-cookie',
+        blacklist_endpoint='foo',
+    )
+
+
+def test_decode_jwt(jwt_manager, jwt_priv_key):
+    payload = {'foo': 'bar'}
+    encoded = jwt.encode(payload, jwt_priv_key, 'RS256')
+    assert jwt_manager._decode_jwt(encoded) == payload
```

Comment:
This might be a good candidate for a `hypothesis` test, although since it's just doing some error handling around a library call it might not be *thaaat* useful.

```python3
@given(payload=st.dictionaries(...))
def test_jwt_encoding(payload, jwt_pub_key, jwt_priv_key):
    jwt_manager = JwtManager(... keys and stuff here ...)
    assert jwt_manager._decode_jwt(jwt_manager._encode_jwt(payload)) == payload
```

You wouldn't be able to use the `jwt_manager` fixture because hypothesis doesn't like you using function scoped fixtures since they wont be reset between calls.